samples/linker-instructions.md

```diff
@@ -0,0 +1,64 @@
+# Using the .NET IL Linker
+
+The .NET team has built a linker to reduce the size of applications. In trivial cases, it can reduce the size of applications by 50%. The size wins may be more favorable or more moderate for larger applications.
+
+The linker currently only supports publishing self-contained applications. It will fail unless you specify a runtime ID. It is currently an experimental feature. We intend to productize it in a subsequent .NET Core release.
```

Comment:
I think it is helpful to describe very briefly what the linker does -- remove unreachable definitions -- at this stage. This will help users understand the later usage such as "linker is too aggressive" later